src/diff/props.js

```diff
@@ -66,7 +66,7 @@ function setProperty(dom, name, value, oldValue, isSvg) {
 		return;
 	}
 	// Benchmark for comparison: https://esbench.com/bench/574c954bdb965b9a00965ac6
-	else if (name[0]==='o' && name[1]==='n') {
+	else if (name[0]==='o' && name[1]==='n' && (typeof value==='function' || typeof oldValue==='function')) {
```

Comment:
Maybe we should add some tests for this specific handler? Maybe there are others?

Reply:
Yes, was planning on it but drafted this to see if this was the way to go, like this we could allow people to insert props starting with on

Reply:
Drafted some tests for this, thing is that the preact changes excludes the need for this change, I just don't know which is better personally.

Reply:
I don't think these checks belong into core here. They're not useful in production and meant for development purposes only. The code above in `preact/debug` already does that :+1:

Reply:
So just limit it to debug then?

Reply:
yup :+1: